### PR TITLE
Add tvOS and watchOS support

### PR DESCRIPTION
**Pull request by mkhon**
Tested with the following targets:
- iOS simulator
- iOS generic device
- tvOS simulator
- tvOS generic device

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -61,4 +77,4 @@ The approach to framework packaging is drawn from `Jeff Verkoeyen`_, and
 `Ernesto García's`_ tutorials.
 
 .. _Jeff Verkoeyen: https://github.com/jverkoey/iOS-Framework
-.. _Ernesto García's: http://www.raywenderlich.com/41377/creating-a-static-library-in-ios-tutorial
+.. _Ernesto García1G's: http://www.raywenderlich.com/41377/creating-a-static-library-in-ios-tutorial
```
Comment (freakboy3742) [inline README.rst]: Is this change intentional?

File: README.rst [hunk 2]
```diff
@@ -1,3 +1,19 @@
+This is a fork of https://github.com/pybee/Python-iOS-support/ that adds
```
Comment (freakboy3742) [inline README.rst]: It won't be a fork if it's merged :-)

File: .gitignore [hunk 3]
```diff
@@ -5,4 +5,4 @@ archive/*
 build/*
 downloads/*
 diff/*
-dist/*
+Python-*-*-support.*.tar.gz
```
Comment (freakboy3742) [inline .gitignore]: Why take things out of the dist directory?

File: README.rst (hunk 1 above)
Comment (mkhon) [inline README.rst]: Ooops. Reverted.

**Comment by freakboy3742** [comment]
This looks pretty good - thanks for the contribution. I've dropped a couple of pieces of feedback on the patch, and I still need to take it for a spin to confirm the bitcode inclusions don't break anything on the iOS side.

File: .gitignore (hunk 3 above)
Comment (mkhon) [inline .gitignore]: I can put tarballs back to dist/ but .framework need to be in build/<os> so that Python build finds them.

File: README.rst (hunk 2 above)
Comment (mkhon) [inline README.rst]: Fixed.

File: .gitignore (hunk 3 above)
Comment (mkhon) [inline .gitignore]: Please also note that Python build is now entirely in the root Makefile so the patch chunk for iOS/Makefile is no longer needed. I decided to not remove it to keep the original patch intact.

Comment (mkhon) [inline .gitignore]: I moved tarballs back to dist/

Comment (mkhon) [inline .gitignore]: I also reverted the change to Python.patch that adds -fembed-bitcode now that Python build is done off the root Makefile.

**Comment by Ahti** [comment]
i've tried this yesterday and while `make all-iOS` works (haven't tested the built framework yet, though), `make all-tvOS` fails after a short bit. This is the output:

```bash
# Unpack sources
mkdir -p build/tvOS/openssl-1.0.2d-appletvsimulator.x86_64
tar zxf downloads/openssl-1.0.2d.tgz --strip-components 1 -C build/tvOS/openssl-1.0.2d-appletvsimulator.x86_64
# Patch apps/speed.c to not use fork() since it's not available on tvOS
sed -ie 's/define HAVE_FORK 1/define HAVE_FORK 0/' build/tvOS/openssl-1.0.2d-appletvsimulator.x86_64/apps/speed.c
# Patch Configure to build for tvOS or watchOS, not iOS
LANG=C sed -ie 's/-D_REENTRANT:iOS/-D_REENTRANT:tvOS/' build/tvOS/openssl-1.0.2d-appletvsimulator.x86_64/Configure
sed: RE error: illegal byte sequence
make: *** [build/tvOS/openssl-1.0.2d-appletvsimulator.x86_64/Makefile] Error 1
```

**Comment by mkhon** [comment]
Ahti, please check the latest fix to use LC_ALL instead of LANG.

**Comment by Ahti** [comment]
Building for tvOS now worked :+1:. Again, I have not tested the built framework yet.

**Comment by freakboy3742** [comment]
Apologies for the delay - I'll make a couple of minor tweaks after merging, but this is great stuff, especially the Makefile re-organization.
